java-frontend/src/main/java/org/sonar/java/se/SymbolicExecutionVisitor.java

```diff
@@ -60,20 +61,61 @@ public void visitNode(Tree tree) {
   @CheckForNull
   public MethodBehavior execute(MethodTree methodTree) {
     try {
-      MethodBehavior methodBehavior = behaviorCache.get(methodTree.symbol());
-      if(methodBehavior == null) {
-        methodBehavior = new MethodBehavior(methodTree.symbol());
-        behaviorCache.put(methodTree.symbol(), methodBehavior);
-        ExplodedGraphWalker walker = egwFactory.createWalker(this);
-        methodBehavior = walker.visitMethod(methodTree, methodBehavior);
-        methodBehavior.completed();
-      }
+      MethodBehavior methodBehavior = new MethodBehavior(methodTree.symbol());
+      behaviorCache.add(methodTree.symbol(), methodBehavior);
+      ExplodedGraphWalker walker = egwFactory.createWalker(behaviorCache);
+      methodBehavior = walker.visitMethod(methodTree, methodBehavior);
+      methodBehavior.completed();
       return methodBehavior;
     } catch (ExplodedGraphWalker.MaximumStepsReachedException | ExplodedGraphWalker.ExplodedGraphTooBigException | BinaryRelation.TransitiveRelationExceededException exception) {
       LOG.debug("Could not complete symbolic execution: ", exception);
     }
     return null;
   }
 
+  class BehaviorCache {
+    final Map<Symbol.MethodSymbol, MethodBehavior> behaviors = new LinkedHashMap<>();
+
+    void add(Symbol.MethodSymbol symbol, MethodBehavior behavior) {
+      behaviors.put(symbol, behavior);
+    }
+
+    public MethodBehavior get(Symbol.MethodSymbol symbol) {
+      if (!behaviors.containsKey(symbol)) {
+        if (isObjectsRequireNonNullMethod(symbol)) {
+          behaviors.put(symbol, createRequireNonNullBehavior(symbol));
+        } else {
+          MethodTree declaration = symbol.declaration();
+          if (declaration != null) {
+            SymbolicExecutionVisitor.this.execute(declaration);
+          }
+        }
+      }
+      return behaviors.get(symbol);
+    }
+
+    private boolean isObjectsRequireNonNullMethod(Symbol symbol) {
+      return symbol.owner().type().is("java.util.Objects") && "requireNonNull".equals(symbol.name());
```

Comment:
Because this requires to refactor methodMatcher to match on symbols which is not available now. Let's implement a bit more of required methods before going for change on MethodMatcher